### ISSUE TITLE
Determine what is the standard of excellent training

### ISSUE THREAD

**Issue by xldsdr124** [nagadomi/waifu2x#356]
In the process of AI training, I found that the value of "learning_rate" is declining continuously, and I don’t understand what is the standard of good judgment for training.

**Comment by xldsdr124** [comment]
These are some valid outputs:

learning rate: 6.5909583485014e-05
{
   loss: 0.00014185911949735
   MSE: 0.00027230762299661
   PSNR: 35.889340583873
}

**Comment by nagadomi** [comment]
The training process minimizes loss.
PSNR is a common image quality metric. The larger it is, the better.
https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio

**Comment by xldsdr124** [comment]
I found that a lot of details were lost during the training of noise reduction level 3. I used many similar noise-free original images as training materials, but after a long period of training, the learning rate has been as low as 1.9746390125668e-06, and the details are still not improved. For the same picture, compared with the noise reduction level 3 of the cnet model, the noise reduction level 3 of the cnet retains more details. Will repeated training improve the level of detail?

**Comment by nagadomi** [comment]
There is a trade-off between denoising and preserving details.
A smaller value for `-nr_rate`(default 0.65) preserves more detail.
However, I have specified `-nr_rate 1` for training noise3 because noise3 doesn't need to preserve the details I think.

The other reason that comes to mind is that I mix the following dot pattern in the training data, about 2% of the total. It helps to train to preserve the details.
![1](https://user-images.githubusercontent.com/287255/91901800-b06abe80-ecdb-11ea-8ab5-0e5f1197a28c.png)
This dot pattern can be generated by the following command.
```
mkdir -p ./dots
th image_generators/dots/gen.lua -n 128 -o ./dots 
```
